crates/turborepo-lib/src/package_changes_watcher.rs

```diff
@@ -167,121 +228,133 @@ impl Subscriber {
-                                    );
+                    match self.initialize_repo_state().await {
+                        Some((new_repo_state, new_gitignore)) => {
+                            repo_state = new_repo_state;
+                            root_gitignore = new_gitignore;
+                            change_mapper = match repo_state.get_change_mapper() {
+                                Some(change_mapper) => change_mapper,
+                                None => {
+                                    break;
                                 }
-                            }
-                            Err(err) => {
-                                // Log the error, rediscover the packages and try again
-                                tracing::error!("error: {:?}", err);
-
-                                let _ = self
-                                    .package_change_events_tx
-                                    .send(PackageChangeEvent::Rediscover);
-                                match self.initialize_repo_state().await {
-                                    Some(new_repo_state) => {
-                                        repo_state = new_repo_state;
-                                        change_mapper = match repo_state.get_change_mapper() {
-                                            Some(change_mapper) => change_mapper,
-                                            None => {
-                                                break;
-                                            }
-                                        }
-                                    }
+                            };
+                        }
+                        None => {
+                            break;
+                        }
+                    }
+                    continue;
+                };
+
+                let gitignore_path = self.repo_root.join_component(".gitignore");
+                if trie.get(gitignore_path.as_std_path()).is_some() {
+                    let (new_root_gitignore, _) = Gitignore::new(&gitignore_path);
+                    root_gitignore = new_root_gitignore;
+                }
+
+                let changed_files: HashSet<_> = trie
+                    .keys()
+                    .filter_map(|p| {
+                        let p = AbsoluteSystemPathBuf::try_from(p.as_path()).ok()?;
+                        self.repo_root.anchor(p).ok()
+                    })
+                    .filter(|p| {
+                        // If in .gitignore or in .git, filter out
+                        !(ancestors_is_ignored(&root_gitignore, p) || is_in_git_folder(p))
+                    })
+                    .collect();
+
+                if changed_files.is_empty() {
+                    continue;
+                }
+
+                let changed_packages = change_mapper.changed_packages(changed_files.clone(), None);
+
+                tracing::warn!("changed_files: {:?}", changed_files);
```

Comment:
`warn` is probably not the right level semantically, but `debug` has so much stuff in it and we don't log it out by default